services/graph/pkg/config/config.go

```diff
@@ -45,51 +45,51 @@ type Spaces struct {
 
-	EducationResourcesEnabled bool `yaml:"education_resources_enabled" env:"LDAP_EDUCATION_RESOURCES_ENABLED;GRAPH_LDAP_EDUCATION_RESOURCES_ENABLED" desc:"Enable LDAP support for managing education related resources"`
+	EducationResourcesEnabled bool `yaml:"education_resources_enabled" env:"OCIS_LDAP_EDUCATION_RESOURCES_ENABLED;LDAP_EDUCATION_RESOURCES_ENABLED;GRAPH_LDAP_EDUCATION_RESOURCES_ENABLED" desc:"Enable LDAP support for managing education related resources" deprecationVersion:"3.0" removalVersion:"3.1" deprecationInfo:"LDAP_EDUCATION_RESOURCES_ENABLED changing name for consistency" deprecationReplacement:"OCIS_LDAP_EDUCATION_RESOURCES_ENABLED"`
 	EducationConfig           LDAPEducationConfig
 }
 
 // LDAPEducationConfig represents the LDAP configuration for education related resources
 type LDAPEducationConfig struct {
-	SchoolBaseDN      string `yaml:"school_base_dn" env:"LDAP_SCHOOL_BASE_DN;GRAPH_LDAP_SCHOOL_BASE_DN" desc:"Search base DN for looking up LDAP schools."`
-	SchoolSearchScope string `yaml:"school_search_scope" env:"LDAP_SCHOOL_SCOPE;GRAPH_LDAP_SCHOOL_SEARCH_SCOPE" desc:"LDAP search scope to use when looking up schools. Supported scopes are 'base', 'one' and 'sub'."`
+	SchoolBaseDN      string `yaml:"school_base_dn" env:"OCIS_LDAP_SCHOOL_BASE_DN;LDAP_SCHOOL_BASE_DN;GRAPH_LDAP_SCHOOL_BASE_DN" desc:"Search base DN for looking up LDAP schools." deprecationVersion:"3.0" removalVersion:"3.1" deprecationInfo:"LDAP_SCHOOL_BASE_DN changing name for consistency" deprecationReplacement:"OCIS_LDAP_SCHOOL_BASE_DN"`
+	SchoolSearchScope string `yaml:"school_search_scope" env:"OCIS_LDAP_SCHOOL_SCOPE;LDAP_SCHOOL_SCOPE;GRAPH_LDAP_SCHOOL_SEARCH_SCOPE" desc:"LDAP search scope to use when looking up schools. Supported scopes are 'base', 'one' and 'sub'." deprecationVersion:"3.0" removalVersion:"3.1" deprecationInfo:"LDAP_SCHOOL_SCOPE changing name for consistency" deprecationReplacement:"OCIS_LDAP_SCHOOL_SCOPE"`
 
-	SchoolFilter      string `yaml:"school_filter" env:"LDAP_SCHOOL_FILTER;GRAPH_LDAP_SCHOOL_FILTER" desc:"LDAP filter to add to the default filters for school searches."`
+	SchoolFilter      string `yaml:"school_filter" env:"OCIS_LDAP_SCHOOL_FILTER;LDAP_SCHOOL_FILTER;GRAPH_LDAP_SCHOOL_FILTER" desc:"LDAP filter to add to the default filters for school searches." deprecationVersion:"3.0" removalVersion:"3.1" deprecationInfo:"LDAP_SCHOOL_FILTER changing name for consistency" deprecationReplacement:"OCIS_LDAP_SCHOOL_FILTER"`
 	SchoolObjectClass string `yaml:"school_objectclass" env:"LDAP_SCHOOL_OBJECTCLASS;GRAPH_LDAP_SCHOOL_OBJECTCLASS" desc:"The object class to use for schools in the default school search filter."`
 
 	SchoolNameAttribute   string `yaml:"school_name_attribute" env:"LDAP_SCHOOL_SCHEMA_SCHOOL_NAME;GRAPH_LDAP_SCHOOL_NAME_ATTRIBUTE" desc:"LDAP Attribute to use for the name of a school."`
```

Comment:
```suggestion
	SchoolFilter      string `yaml:"school_filter" env:"GRAPH_LDAP_SCHOOL_FILTER" desc:"LDAP filter to add to the default filters for school searches."`
	SchoolObjectClass string `yaml:"school_objectclass" env:"GRAPH_LDAP_SCHOOL_OBJECTCLASS" desc:"The object class to use for schools in the default school search filter."`

	SchoolNameAttribute   string `yaml:"school_name_attribute" env:"GRAPH_LDAP_SCHOOL_NAME_ATTRIBUTE" desc:"LDAP Attribute to use for the name of a school."`
```
The `LDAP_` global variant is not necessary because this envvar is only used in the GRAPH sevice. No deprecation necessary as this is very specific and introduced not long time ago and the yaml keys stay the same.